platform/migration/platform-migratable/src/main/java/org/codice/ddf/migration/impl/PlatformMigratable.java

```diff
@@ -0,0 +1,216 @@
+
+    private final Path ddfHome;
+    
+    public PlatformMigratable(@NotNull Path ddfHome) {
+        notNull(ddfHome, "ddfHome cannot be null");
+        this.ddfHome = ddfHome;
+    }
+    
+    @Override
+    @NotNull
+    public MigrationMetadata export(@NotNull Path exportPath) throws MigrationException {
+        notNull(exportPath, "exportPath cannot be null");
+        exportSystemFiles(exportPath);
+
+        Collection<MigrationWarning> migrationWarnings = new ArrayList<>();
+        migrationWarnings.addAll(exportSecurity(exportPath));
+        return new MigrationMetadata(migrationWarnings);
+    }
+
+    @Override
+    @NotNull
+    public String getDescription() {
+        return "Exports all Platform configurations and system files";
+    }
+
+    @Override
+    public boolean isOptional() {
+        return false;
+    }
+    
+    private Collection<MigrationWarning> exportSecurity(Path exportDirectory) {
+        exportDirectory(exportDirectory, WS_SECURITY_DIR);
+
+        Collection<MigrationWarning> migrationWarnings = new ArrayList<>();
+        migrationWarnings.addAll(exportKeystores(exportDirectory));
+        return migrationWarnings;
+    }
+
+    private void exportSystemFiles(Path exportDirectory) {
+        copyFile(ddfHome.resolve(SYSTEM_PROPERTIES), exportDirectory.resolve(SYSTEM_PROPERTIES));
+        copyFile(ddfHome.resolve(USERS_PROPERTIES), exportDirectory.resolve(USERS_PROPERTIES));
+    }
+
+    private Collection<MigrationWarning> exportKeystores(Path exportDirectory) {
+        Collection<MigrationWarning> migrationWarnings = new ArrayList<>();
+        migrationWarnings.addAll(exportExternalFile(exportDirectory, KEYSTORE_SYSTEM_PROP));
+        migrationWarnings.addAll(exportExternalFile(exportDirectory, TRUSTSTORE_SYSTEM_PROP));
+        return migrationWarnings;
+    }
+
+    private Collection<MigrationWarning> exportExternalFile(Path exportDirectory,
+            String propertyWithPath) {
+        Collection<MigrationWarning> migrationWarnings = new ArrayList<>();
+        String keystore = getProperty(propertyWithPath);
+        migrationWarnings.addAll(checkIfPathIsMigratable(propertyWithPath, Paths.get(keystore)));
+        if (migrationWarnings.isEmpty()) {
+            Path source = ddfHome.resolve(keystore);
+            copyFile(source, constructDestination(source, exportDirectory));
+        }
+        return migrationWarnings;
+    }
+
+    private Path constructDestination(Path pathToExport, Path exportDirectory) {
+        Path destination = exportDirectory.resolve(ddfHome.relativize(pathToExport));
+        return destination;
+    }
+
+    private void exportDirectory(Path destinationRoot, String directoryToCopy) {
+        Path source = ddfHome.resolve(Paths.get(directoryToCopy));
+        copyDirectory(source, constructDestination(source, destinationRoot));
+    }
+
+    private String getProperty(String property) throws MigrationException {
```

Comment:
Some of this has been moved to the MigratableUtil class. I also created an AbstractMigratable in the api